### ISSUE TITLE
[REVIEW]: AuditoryStimuli.jl: A Julia package for generating real-time auditory stimuli

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3613]
**Submitting author:** @rob-luke (<a href="http://orcid.org/0000-0002-4930-8351">Robert Luke</a>)
**Repository:** <a href="https://github.com/rob-luke/AuditoryStimuli.jl" target ="_blank">https://github.com/rob-luke/AuditoryStimuli.jl</a>
**Version:** v0.0.11
**Editor:** @bmcfee
**Reviewer:** @vahid-sb, @jongwook
**Archive:** <a href="https://doi.org/10.5281/zenodo.5525443" target="_blank">10.5281/zenodo.5525443</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/b91a016d27744caa7bce0ad558c177ba/status.svg)](https://joss.theoj.org/papers/b91a016d27744caa7bce0ad558c177ba)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/b91a016d27744caa7bce0ad558c177ba"><img src="https://joss.theoj.org/papers/b91a016d27744caa7bce0ad558c177ba/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/b91a016d27744caa7bce0ad558c177ba/status.svg)](https://joss.theoj.org/papers/b91a016d27744caa7bce0ad558c177ba)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/rob-luke/AuditoryStimuli.jl" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@vahid-sb & @jongwook, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @bmcfee know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @vahid-sb
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/rob-luke/AuditoryStimuli.jl">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/rob-luke/AuditoryStimuli.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@rob-luke) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @jongwook
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/rob-luke/AuditoryStimuli.jl">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/rob-luke/AuditoryStimuli.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@rob-luke) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @vahid-sb, @jongwook it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.02 s (1535.6 files/s, 109903.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Julia                           14            383             79           1135
Markdown                        12            194              0            507
YAML                             7              9              9            143
TOML                             2              4              0             42
-------------------------------------------------------------------------------
SUM:                            35            590             88           1827
-------------------------------------------------------------------------------


Statistical information for the repository '058112c24edfc33faa47f0ec' was
gathered on 2021/08/13.
No commited files with the specified extensions were found.
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3613 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by rob-luke** [comment]
@whedon generate pdf from branch manuscript

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch manuscript. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03613/joss.03613/10.21105.joss.03613.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03613/joss.03613/10.21105.joss.03613.pdf) :page_facing_up: :point_left:

**Comment by jongwook** [comment]
@whedon list reviewers

**Comment by whedon** [comment]
Here's the current list of reviewers: https://bit.ly/joss-reviewers

**Comment by jongwook** [comment]
Great work @rob-luke! Below are my reviews and suggestions for minor revisions.

- The paper is written very well, succinctly and effectively conveying how the AuditoryStimuli.jl package can be a valuable addition as a real-time auditory signal synthesis tool, where it bridges the efficiency required for real-time applications and the ease of use of Julia programming language.
- Some of the URLs in the citation stick out of the margin and overflow out of the page. I didn't see this issue in a few of JOSS paper I've checked, but I'm not sure this only happens in the draft mode.
- The documentation is very well-written as well, complete with installation instructions, introductory usage, and detailed examples for the various features that the package provides. The documentation website is automatically built and deployed by Documenter.jl.
- Quite a shameless plug, but being a Julia library on audio, [MusicProcessing.jl](https://github.com/JuliaMusic/MusicProcessing.jl) could be considered as a citation to mention alongside librosa. The [publication](https://www.aes.org/e-lib/browse.cfm?elib=18769) is not open-access, but happy to share the preprint.
- The repository is quite mature, with about 3 years in development, and every commit is validated with automated tests, which currently shows 100% code coverage. The Julia code files amount to about 1700 lines of code, ~40% of which is for tests.
- The git repository takes up 200M after the initial clone, which doesn't seem to be used in the `HEAD` of any of the branches. In case those are taken up by a few large files, stripping them from the Git history using `git filter-branch` commands would be helpful
- The GitHub repository currently lacks the [community guidelines](https://joss.readthedocs.io/en/latest/review_criteria.html#community-guidelines), which is one of the JOSS review criteria. It could be a few sentences in the `README.md` welcoming contribution, and addition of `CONTRIBUTING.md` or issue templates would be a great plus, which can show up when someone opens up an issue.

**Comment by rob-luke** [comment]
Thank you for the review and kind words @jongwook. I will briefly respond to your feedback. Please let me know if you want more information or further discussion. I also welcome any additional feedback once the review is done too.

> Some of the URLs in the citation stick out of the margin and overflow out of the page. I didn't see this issue in a few of JOSS paper I've checked, but I'm not sure this only happens in the draft mode.

I confirm that I see this too. I looked up the three most recent JOSS papers and I confirm that they do not have this URL formatting problem. I then looked up their .bib files and they look the same as mine (see figure below, the URL string is very long). So I am not sure how to solve this @bmcfee do you have any suggestions as to how this may be solved?

Figure from Vaidya et al., (2021). JPhyloRef: a tool for testing and resolving phyloreferences. Journal of Open Source Software, 6(64), 3374, https://doi.org/10.21105/joss.03374
![image](https://user-images.githubusercontent.com/748691/130060503-fe7faf51-f4d7-4f4f-88f5-4b8a0f847030.png)

> Quite a shameless plug, but being a Julia library on audio, MusicProcessing.jl could be considered as a citation to mention alongside librosa. The publication is not open-access, but happy to share the preprint.

A completely appropriate plug and I should have added this earlier as I am aware of the great package, but didnt know it had an associated article. I have updated the manuscript to include this reference.

> 40% of which is for tests

Yep, I love tests.

> The git repository takes up 200M after the initial clone, which doesn't seem to be used in the HEAD of any of the branches. In case those are taken up by a few large files, stripping them from the Git history using git filter-branch commands would be helpful

Thanks for the tip. It was the documentation taking up all the space. I have cleared that history and now the repository is <800 KiB.

> The GitHub repository currently lacks the community guidelines, which is one of the JOSS review criteria. It could be a few sentences in the README.md welcoming contribution, and addition of CONTRIBUTING.md or issue templates would be a great plus, which can show up when someone opens up an issue.

As per your suggestion I have added community guidelines to the README. I have also added a 
CONTRIBUTING.md file. And I have created issue templates.


Again, thank you @jongwook for the review.

**Comment by rob-luke** [comment]
@whedon generate pdf from branch manuscript

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch manuscript. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03613/joss.03613/10.21105.joss.03613.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03613/joss.03613/10.21105.joss.03613.pdf) :page_facing_up: :point_left:

**Comment by bmcfee** [comment]
> So I am not sure how to solve this @bmcfee do you have any suggestions as to how this may be solved?

This is apparently a templating issue on our side -- don't worry about it for now, we'll take care of it when it comes time for the final version.

**Comment by jongwook** [comment]
@rob-luke Thanks for responding so fast, and I meant "40% of which is for tests" as a totally positive note as well!

I confirm that the git clone size has been reduced, and the community guidelines and the issue templates look great. Thanks!

@bmcfee All checkboxes marked off!

**Comment by rob-luke** [comment]
Thanks @bmcfee and @jongwook

**Comment by vahid-sb** [comment]
Hi @rob-luke. My first comment is that please add installation notes to the documentation.

**Comment by rob-luke** [comment]
> Hi @rob-luke. My first comment is that please add installation notes to the documentation. 

As @jongwook notes "The documentation is very well-written as well, complete with installation instructions". Are the installation instructions are the first subtopic on the documentation page. See https://rob-luke.github.io/AuditoryStimuli.jl/dev/#Installation

This is common practice in Julia packages. What step is missing from these instructions that you would like me to add?

**Comment by vahid-sb** [comment]
@rob-luke the dependencies for your package. I had to clone and install `MusicProcessing` from the disk. And it seems I have to do the same for `MP3` package too. Neither can be found in the registry, hence the example you have added cannot be run without extra steps. Also, the `MP3` package doc says it is not maintained anymore. Is there no other package in the registry that can load mp3 files?

**Comment by rob-luke** [comment]
Hi @vahid-sb, where have you seen `MusicProcessing` in relation to this software package? It is unrelated to my software and review. I searched the repository for this term and it does not appear. Where did you get the impression it was required? Similarly the MP3 package is not utilised in my code or even discussed in the documentation.

The WAV package is utilised in the examples for this package but is not a core requirement. As such it is listed in the file `docs/Project.toml`, as per Julia standards. I am reasonably confident (but happy to be corrected) that all necessary packages are listed in the appropriate location as the continuous integration tests are passing and the documentation and the examples is also being built by the continuous integration each time code is modified.

I would be pleased to get to the cause of this confusion with you, how can I help? You can see the requirement including version restrictions at https://github.com/rob-luke/AuditoryStimuli.jl/blob/main/Project.toml and the extended requirements for the documentation (includes additional packages such as the documentation building software and useful tips for users) at https://github.com/rob-luke/AuditoryStimuli.jl/blob/main/docs/Project.toml

**Comment by vahid-sb** [comment]
@rob-luke oh man! My bad! It was a mix-up with the other package the other reviewer mentioned in the comments. Things got mixed up in many many open tabs. My apologies. I will delete the last few comments, since I see they are irrelevant now.

**Comment by rob-luke** [comment]
No worries. I'm glad we got to the bottom of it quickly and didn't use to much of your time

**Comment by rob-luke** [comment]
Don't bother about deleting comments. No worries, I understand we are all busy. I don't want to use more of your time than required cleaning a chat thread.

**Comment by vahid-sb** [comment]
@rob-luke I'd come back to you with my comments as soon as I can.

**Comment by rob-luke** [comment]
> @rob-luke I'd come back to you with my comments as soon as I can. 

Thanks @vahid-sb I appreciate your volunteered time.

**Comment by bmcfee** [comment]
Just giving another gentle nudge to @vahid-sb here - any update on when you'll be able to move this forward would be appreciated!

**Comment by vahid-sb** [comment]
Hi @bmcfee 
I haVe tested the code, run the example files and looked into the documentation. The code does what the documentation says it does and I find it comprehensive enough to be suitable for JOSS. As somebody who has worked on behavioural studies, I see there are some features that I might have needed and currently not available. But I do not find it reasonable to ask for all the features that various users may or may not need, at least in this stage of development. Hence, I'm ready to approve this software/paper in its current form to be published in JOSS.

**Comment by rob-luke** [comment]
Wonderful!! Thanks @vahid-sb 

Of course feel free to add any wish-list features as an issue on the repo. I plan to keep developing this software and adding more features as required.

**Comment by vahid-sb** [comment]
@rob-luke I just ticked all the boxes. You have done a good job and I hope it will be used wildly by experimentalists and hope to see it constantly improving. Although I should add given what I have seen, wouldn't bet on it. But maybe efforts such as yours would help to introduce Julia-based libraries to a wider audience.

**Comment by vahid-sb** [comment]
@bmcfee Thanks for the invitation for the review. It was an interesting experience.

**Comment by bmcfee** [comment]
Thanks @vahid-sb ! It does look like you still have an unchecked box (Performance) -- can you resolve that?

**Comment by vahid-sb** [comment]
@bmcfee done!

**Comment by bmcfee** [comment]
@rob-luke back to the bib formatting issues, can you try using `url={...}` instead of `howpublished={...}`?  It may not resolve the issue entirely, but it's the first thing to try.

**Comment by WilkAndy** [comment]
Somehow i got subscribed to this, perhaps because i just agreed to review another JOSS paper.  I am now unsubscribing.  If i need to be subscribed, one of the editors/authors will have my email address and can email me.

**Comment by rob-luke** [comment]
@whedon generate pdf from branch manuscript

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch manuscript. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3613 with the following error: 
```
 Error producing PDF.
! TeX capacity exceeded, sorry [input stack size=5000].
\reserved@a ->\def \reserved@a 
                               *{\@hspacer }\reserved@a 
l.504 ...}https://github.com/sam81/pychoacoustics}

Looks like we failed to compile the PDF

```

**Comment by rob-luke** [comment]
@whedon generate pdf from branch manuscript

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch manuscript. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03613/joss.03613/10.21105.joss.03613.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03613/joss.03613/10.21105.joss.03613.pdf) :page_facing_up: :point_left:

**Comment by rob-luke** [comment]
Genius @bmcfee!! That seems to have fixed the issue. Anything else I need to do now the reviews are in?

**Comment by rob-luke** [comment]
Actually I can see that the patent numbers aren't formatting correctly in the bibliography either. Please give me a few hours to try and fix this in the bib file.

**Comment by rob-luke** [comment]
@whedon generate pdf from branch manuscript

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch manuscript. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03613/joss.03613/10.21105.joss.03613.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03613/joss.03613/10.21105.joss.03613.pdf) :page_facing_up: :point_left:

**Comment by rob-luke** [comment]
Thanks for your patience @bmcfee. I had to switch from `@misc` to `@patent` for the formatting to work. But I think it all looks good now. Let me know if there's anything else I need to do.

**Comment by bmcfee** [comment]
@whedon check references

**Comment by rob-luke** [comment]
Maybe I can help out our overworked friend Whedon.

There are no dois for the software references, so I added urls:
* https://github.com/sam81/pychoacoustics
* https://github.com/spatialaudio/python-sofa
* https://github.com/JuliaAudio/SampledSignals.jl

There are no dois for the patents, so I added URLs to the google patent pages:
* https://patents.google.com/patent/US10412518B2/en
* https://patents.google.com/patent/US10812889B2/en
* https://patents.google.com/patent/US10504537B2/en

I could not find a doi for:
* Fast music and audio processing using the Julia language by @jongwook 
  https://www.aes.org/e-lib/browse.cfm?elib=18769
  There is a DOI for the whole proceedings (https://doi.org/10.17743/aesconf.2017.978-1-942220-15-2) should I use that?

I dont know much about the doi system, so please correct any misunderstanding I have.

**Comment by bmcfee** [comment]
Thanks @rob-luke - I think that should be fine from the reference side.

You are right that there does not appear to be a DOI for that particular reference. I think it's okay to leave it as is; citing the entire proceedings would be confusing.

Some last minor comments on the article from my side:

- End of paragraph 1 is a sentence fragment
- Line 35: "users responses" => "users' responses" or "user's responses"?  Although, in context, "subject" might be more appropriate than "user" since the latter refers to the experimenter/researcher later in the article.  I'll defer to your judgment on that.
- Line 82: "researchs" => "researchers"

Otherwise, the last steps here are to push an up-to-date version of the software (in case anything has changed in response to the reviews); tag the github repo accordingly, and archive it to zenodo or figshare to get a DOI.  When you've done that, comment back with the software's DOI so we can link it to the paper.

**Comment by rob-luke** [comment]
Thanks for the review @bmcfee. I removed the entire sentence fragment as the same content was addressed in the final paragraph. I changed `users responses` to `participant's responses`, and fixed the typo you found. Thanks for the extra set of eyes!

I just released a new version of the software. And also created a Zenodo doi `10.5281/zenodo.5525443`.

**Comment by bmcfee** [comment]
@whedon generate pdf from branch manuscript

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch manuscript. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03613/joss.03613/10.21105.joss.03613.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03613/joss.03613/10.21105.joss.03613.pdf) :page_facing_up: :point_left:

**Comment by bmcfee** [comment]
@whedon set 10.5281/zenodo.5525443 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5525443" target="_blank">10.5281/zenodo.5525443</a> is the archive.

**Comment by bmcfee** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3613 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by bmcfee** [comment]
Thanks for the fixes @rob-luke -- everything's looking good!

**Comment by bmcfee** [comment]
@whedon recommend-accept from branch manuscript

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1117/1.NPh.8.2.025008 is OK
- 10.21105/joss.03284 is OK
- 10.3758/s13428-018-01193-y is OK
- 10.25080/majora-7b98e3ed-003 is OK
- 10.1016/j.neuroimage.2013.10.027 is OK
- 10.1155/2011/156869 is OK
- 10.1137/141000671 is OK
- 10.1109/TNSRE.2016.2551302 is OK
- 10.1016/j.neuroimage.2018.01.033 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2607

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2607, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch manuscript 
```

**Comment by kthyng** [comment]
Hi @rob-luke! Final steps to wrap up here. Can you update the Zenodo archive metadata so the title and author list exactly match the JOSS paper?

**Comment by kthyng** [comment]
Paper comments:
* first page, "existing tools such as (Carcagno 2012)" — this reference should be inline, not parenthetical. You can make this change by removing the [] around the reference.
* final sentence in that paragraph — don't start a sentence with "i.e", at least write out as 'That is' or change wording.
* Same parenthetical problem with a reference in the "Notes" section.
* fix capitalization in your references, for example "python" and "julia" should be capitalized. You can preserve capitalization in your .bib file by putting {} around strings.

**Comment by rob-luke** [comment]
@whedon generate pdf from branch manuscript

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch manuscript. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03613/joss.03613/10.21105.joss.03613.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03613/joss.03613/10.21105.joss.03613.pdf) :page_facing_up: :point_left:

**Comment by rob-luke** [comment]
Thanks for carefully checking the details @kthyng. I have updated the zenodo title and made the changes to the manuscript and bib that you suggested. Please let me know if there is anything else I should do. Cheers

**Comment by kthyng** [comment]
@whedon accept deposit=true from branch manuscript

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1442491471949488133) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2614
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03613](https://doi.org/10.21105/joss.03613)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by bmcfee** [comment]
Congrats @rob-luke ! :tada: 

And thanks again to reviewers @jongwook and @vahid-sb !

**Comment by kthyng** [comment]
Congrats on your new publication @rob-luke! Many thanks to editor @bmcfee and reviewers @vahid-sb and @jongwook!!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03613/status.svg)](https://doi.org/10.21105/joss.03613)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03613">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03613/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03613/status.svg
   :target: https://doi.org/10.21105/joss.03613
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03613/status.svg)](https://doi.org/10.21105/joss.03613)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by rob-luke** [comment]
Thank you @kthyng @bmcfee, and please keep me in mind as a reviewer in the future (I signed up previously).

Thanks @jongwook and @vahid-sb for the review!
